cartography/intel/aws/s3.py

```diff
@@ -16,16 +16,24 @@
 def get_s3_bucket_list(boto3_session):
     client = boto3_session.client('s3')
     # NOTE no paginator available for this operation
-    return client.list_buckets()
+
+    buckets = client.list_buckets()
+    for bucket in buckets['Buckets']:
+        bucket['Region'] = client.get_bucket_location(Bucket=bucket['Name'])['LocationConstraint']
+    return buckets
 
 
 @timeit
 def get_s3_bucket_details(boto3_session, bucket_data):
     """
     Iterates over all S3 buckets. Yields bucket name (string) and pairs of S3 bucket policies (JSON) and ACLs (JSON)
     """
-    client = boto3_session.client('s3')
+    # client = boto3_session.client('s3')
     for bucket in bucket_data['Buckets']:
+        # Use us-east-1 region if no region was recognized for buckets
+        # It was found that client.get_bucket_location does not return a region for buckets
+        # in us-east-1 region
+        client = boto3_session.client('s3', bucket['Region'])
```

Comment:
I don't think we should create a new boto3 client for every bucket in the environment. Instead, we should call your new `get_s3_bucket_list()`, group up the results by region, and create only as many clients as there are regions to create details. 

Does this make sense? I can help sketch something if this isn't clear.

Reply:
Noted @achantavy . Let me work on this.

Reply:
@achantavy , Instead of grouping buckets, I've created a dictionary that stores s3 clients based on their region in get_s3_bucket_details function. The client is chosen from the store according to the bucket's region. Clients are added to the dictionary only for bucket regions. Kindly review.